src/realm/sync/noinst/server/server.cpp

```diff
@@ -1684,12 +1684,32 @@ class HTTPConnection {
                 value = *sec_websocket_protocol;
             HttpListHeaderValueParser parser{value};
             std::string_view elem;
+            // Support the new (with '#') and old (with '/') protocol prefix strings
+            std::vector<std::string_view> prefixes = {get_pbs_websocket_protocol_prefix(),
+                                                      get_old_pbs_websocket_protocol_prefix()};
+            std::string_view last_prefix;
+            auto check_prefix = [&prefixes, &last_prefix](std::string_view elem) -> std::optional<std::string_view> {
```

Comment:
I'm not a big fan of this lambda. Instead of this, you could just check for both prefixes below where this check is performed:
```
auto prefix = get_pbs_websocket_protocol_prefix();
auto old_prefix = get_old_pbs_websocket_protocol_prefix();
REALM_ASSERT(prefix.size() == old_prefix.size());
bool begins_with = (elem.size() >= prefix.size() && (
                                    std::equal(elem.data(), elem.data() + prefix.size(), prefix.data()) || (std::equal(elem.data(), elem.data() + prefix.size(), old_prefix.data()));
```